R-packages/evalcast/R/download_signals.R

```diff
@@ -31,21 +31,11 @@ download_signal <- function(...) {
     )
   }
   message(msg)
-  ## Why two calls below?? Commented out one
-  ## base::suppressMessages({covidcast_signal(...)})
+  
   out <- base::suppressMessages({covidcast_signal(...)})
   if (args$geo_type == "state") {
-      e  <- new.env(parent = emptyenv())
-      county_geo  <- utils::data("county_geo", package = "covidcast", envir = e)
-      state_fips <- e$county_geo %>%
-          select(.data$STATE, .data$COUNTYNAME, .data$FIPS) %>%
-          mutate(fips = str_sub(.data$FIPS, end = 2),
-                 geo_value = tolower(.data$STATE)) %>%
-          distinct(.data$geo_value, .data$fips)
-    out <- out %>%
-      left_join(state_fips, by = "geo_value") %>%
-      mutate(geo_value = .data$fips) %>%
-      select(-.data$fips)
+    out$geo_value = covidcast::name_to_fips(paste0("^",
```

Comment:
recommend using `fixed = TRUE` in name_to_fips instead of paste0 and regex.

Reply:
Nevermind on the `fixed` recommendation. That still matches on substrings, so adding `^` and `$` looks necessary.

I'll note that `District of Columbia` is the name of both the state and county, which results in a warning but ultimately returns the correct fips value, but is technically reliant on the order in county_census.

Reply:
It's a good thought but unfortunately that doesn't work (I tried it).  Consider:
```
grep("Wyoming", c("Wyoming", "Wyoming County"), fixed = TRUE)
```

Reply:
DC: it's a good point.  I guess we're lucky because the ordering is by increasing FIPS code, so "11000" appears before "11001".  See my next commit for the most robust fix; it fetches all the matches and extracts just the ones that end with "000".